utils/controleval.py

```diff
@@ -49,95 +49,160 @@ def validate_args(ctrlmgr, args):
+    try:
+        return os.listdir(products_dir)
+    except Exception as e:
+        print(e)
+        exit(1)
 
-    if total == 0:
-        print("No controls founds with the given inputs. Maybe try another level.")
+
+def validate_product(product):
+    products = get_available_products()
+    if product not in products:
+        print(f"Error: Product '{product}' is not valid.")
+        print_options(products)
         exit(1)
 
+
+def get_product_dir(product):
+    validate_product(product)
+    return os.path.join(SSG_ROOT, "products", product)
+
+
+def get_product_yaml(product):
+    product_dir = get_product_dir(product)
+    product_yml = os.path.join(product_dir, "product.yml")
+    if os.path.exists(product_yml):
+        return product_yml
+    else:
+        print(f"'{product_yml}' file was not found.")
+        exit(1)
+
+
+def count_implicit_status(ctrls, status_count):
+    status_count['all'] = len(ctrls)
+    status_count['applicable'] = len(ctrls) - status_count[controls.Status.NOT_APPLICABLE]
+    status_count['assessed'] = status_count['applicable'] - status_count[controls.Status.PENDING]
+    status_count['not assessed'] = status_count['applicable'] - status_count['assessed']
+    status_count['full coverage'] = status_count[controls.Status.AUTOMATED]\
+                                    + status_count[controls.Status.DOCUMENTATION]\
+                                    + status_count[controls.Status.INHERENTLY_MET]\
+                                    + status_count[controls.Status.MANUAL]
+    status_count['partial coverage'] = status_count['full coverage']\
+                                       + status_count[controls.Status.PARTIAL]
+    return status_count
+
+
+def create_implicit_control_lists(ctrls, control_list):
+    control_list['all'] = ctrls
+    control_list['applicable'] = ctrls - control_list[controls.Status.NOT_APPLICABLE]
+    control_list['assessed'] = control_list['applicable'] - control_list[controls.Status.PENDING]
+    control_list['not assessed'] = control_list['applicable'] - control_list['assessed']
+    control_list['full coverage'] = ctrls - control_list[controls.Status.DOES_NOT_MEET]\
+                                    - control_list[controls.Status.NOT_APPLICABLE]\
+                                    - control_list[controls.Status.PARTIAL]\
+                                    - control_list[controls.Status.PENDING]\
+                                    - control_list[controls.Status.PLANNED]\
+                                    - control_list[controls.Status.SUPPORTED]
+    control_list['partial coverage'] = ctrls - control_list[controls.Status.DOES_NOT_MEET]\
+                                    - control_list[controls.Status.NOT_APPLICABLE]\
+                                    - control_list[controls.Status.PENDING]\
+                                    - control_list[controls.Status.PLANNED]\
+                                    - control_list[controls.Status.SUPPORTED]
```

Comment:
Could the common items be extracted out to a variable?

Reply:
Good idea. This can indeed be improved. Thanks

Reply:
Done.